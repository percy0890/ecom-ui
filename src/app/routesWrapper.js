import React, { Suspense } from 'react';
import PropTypes from 'prop-types';
import {
  Switch,
  HashRouter as Router,
  Redirect,
  Route,
} from 'react-router-dom';
import { connect } from 'react-redux';
import { compose } from 'redux';
import _forEach from 'lodash/forEach';
import MomentUtils from '@date-io/moment';
import { MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { materialUITheme } from 'app/utils/common';
import ErrorBoundary from './components/ErrorBoundary';

import Loader from './components/Loader';
import SideDrawer from './components/SideDrawer';
import Footer from './components/Footer';
import PageHeader from './components/PageHeader';
import { getHOCWrappedComponent } from './utils/component';
import config from './config/index.config';
import { routeMap } from './app.routes';

const getRouteComponents = routes => {
  const routeComponents = [];
  _forEach(routes, route => {
    routeComponents.push(
      <Route
        key={route.path}
        path={route.path}
        exact={route.exact}
        component={getHOCWrappedComponent(route.screen, route)}
      />,
    );
  });
  return routeComponents;
};

export class Routes extends React.PureComponent {
  render() {
    const { accessToken, siteId } = this.props;
    this.theme = materialUITheme();

    const lazyRoutesMap = (
      <MuiThemeProvider theme={this.theme}>
        <MuiPickersUtilsProvider utils={MomentUtils}>
          <Suspense fallback={<Loader />}>
            <Switch>
              {getRouteComponents(routeMap)}
              <Redirect to={config.PUBLIC_ROOT} /> {/* Redirect if 404 */}
              {/* <Route component={NoMatch} /> IF 404 to be displayed separately */}
            </Switch>
          </Suspense>
        </MuiPickersUtilsProvider>
      </MuiThemeProvider>
    );

    return (
      <Router>
        <ErrorBoundary>
          {/* {accessToken && ( */}
          {accessToken && (
            <div className="layout-container">
              <SideDrawer menuItems={routeMap} />
              <main className="page-container">
                <PageHeader siteId={siteId} />
                {lazyRoutesMap}
              </main>
              <Footer menuItems={routeMap} />
            </div>
          )}
          {!accessToken && lazyRoutesMap}
        </ErrorBoundary>
      </Router>
    );
  }
}

Routes.propTypes = {
  accessToken: PropTypes.string,
  siteId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

const mapStateToProps = state => ({
  accessToken: state.global.userInfo.accessToken,
  siteId: state.global.userInfo.siteId,
});

export default compose(
  connect(
    mapStateToProps,
    null,
  ),
)(Routes);
