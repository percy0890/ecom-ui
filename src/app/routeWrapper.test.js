import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';

import configureStore from '../store';
import { Routes } from './routesWrapper';

describe('<Routes />', () => {
  let component;
  const { store } = configureStore({});

  beforeEach(() => {
    component = shallow(<Routes siteId="1" />);
  });

  it('Routes component renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Routes siteId="1" />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render routes with PageHeader and SideDrawer if accessToken is set', () => {
    component.setProps({ accessToken: 'abcd' });
    component.update();
    const layoutContainer = component.find('.layout-container');
    expect(layoutContainer.length).toBe(1);
  });
});
