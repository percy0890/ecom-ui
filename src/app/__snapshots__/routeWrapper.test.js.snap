// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> Routes component renders correctly 1`] = `
<div
  className="error-page"
>
  <section
    id="not-found"
  >
    <div
      id="title"
    >
      <button
        className="btn btn-secondary goback-btn  undefined "
        onClick={[Function]}
        type="button"
      >
        <div
          className=" btn-content-container "
        >
          <span
            className="label"
          >
            Go to previous page.
          </span>
        </div>
      </button>
    </div>
    <div
      className="circles"
    >
      <p>
        <img
          alt="error-icon"
          src="IMAGE_MOCK"
        />
        <small>
          Oops! - Something went wrong.
        </small>
         
        <br />
        Please try again, or contact the support team for assistance if required.
      </p>
      <span
        className="circle big"
      />
      <span
        className="circle med"
      />
      <span
        className="circle small"
      />
    </div>
  </section>
</div>
`;
